src/super_context.mli

```diff
@@ -197,6 +197,8 @@ module Libs : sig
       all the files with extension [ext] of libraries [libs]. *)
   val file_deps : t -> Lib.L.t -> ext:string -> Path.t list
 
+  val file_deps_with_exts : t -> (Lib.t * string) list -> Path.t list
```

Comment:
A suggestion for a better name/implementation would be appreciated. I could also implement `file_deps` on top of this function, but don't see the point of making the code slower for no reason.

Reply:
The name is not too much of a problem, but I agree that duplicating is not ideal.

Fortunately, you can extract the body of `List.map` into private function `val file_deps_for_lib : Lib.t -> ext:string -> Build_system.t -> Path.t`. That way, the two functions `file_deps` & `file_deps_with_exts` become one-liners.